### ISSUE TITLE
bug: Pre-signed S3 PUT request fails with Cors Error

### ISSUE THREAD

**Issue by richardARPANET** [localstack/localstack#4398]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Similar to this old issue https://github.com/localstack/localstack/issues/2564

With this bucket cors config:

```
aws --endpoint-url=http://localhost:4566 s3api get-bucket-cors --bucket=app-local-media
{
    "CORSRules": [
        {
            "AllowedHeaders": [
                "*"
            ],
            "AllowedMethods": [
                "GET",
                "PUT",
                "POST",
                "HEAD"
            ],
            "AllowedOrigins": [
                "*",
                "http://localhost:8000"
            ],
            "ExposeHeaders": [
                "ETag"
            ],
            "MaxAgeSeconds": 86400
        }
    ]
}
```

When my application (running on localhost:8000 makes the below request using JavaScript, the request fails with:

```
Access to XMLHttpRequest at 'http://localhost:4566/app-local-media/uploads/4sE0zHdESPGt.mp4?uploadId=ivE4E7Ud7YmSy0XqQeAs1Elg8N38MaqS1UmPhN0Qdd3qKxvIh7reUNTlw&partNumber=1&AWSAccessKeyId=test&Signature=%2BmFFaqqI4Rsq905iqDDNb1ke%2FL8%3D&Expires=1627936836' 
from origin 'http://localhost:8000' 
has been blocked by CORS policy: 
Response to preflight request doesn't pass access control check: 
No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

```
curl 'http://localhost:4566/app-local-media/uploads/4sE0zHdESPGt.mp4?uploadId=ivE4E7Ud7YmSy0XqQeAs1Elg8N38MaqS1UmPhN0Qdd3qKxvIh7reUNTlw&partNumber=1&AWSAccessKeyId=test&Signature=%2BmFFaqqI4Rsq905iqDDNb1ke%2FL8%3D&Expires=1627936836' \
  -X 'PUT' \
  -H 'Access-Control-Allow-Origin: *' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Referer: ' \
  -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36' \
  --data-raw $'\u0000...' --compressed
```

### Expected Behavior

**The same application code, with the same configuration steps (they are automated) when pointed at an AWS hosted S3 bucket functions correctly.**

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    
```
  localstack:
    image: localstack/localstack:latest
    container_name: localstack
    ports:
      - "4566:4566"
      - "4571:4571"
      - "8055:8080"
```

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    1. Create bucket
    2. Set cors config
    3. Generate pre-signed upload url
    4. attempt to upload to pre-signed url

### Environment

```markdown
- OS: Ubuntu 20.04
- LocalStack: latest
```

**Comment by richardARPANET** [comment]
This issue is still occurring

**Comment by richardARPANET** [comment]
This issue is still occurring

**Comment by richardARPANET** [comment]
Managed to work around this issue by setting env var `S3_SKIP_SIGNATURE_VALIDATION=0`

**Comment by monkut** [comment]
I'm experiencing the same/similar issue with the latest docker version, `localstack/localstack:0.12.18`.

I was getting the following error with `localstack/localstack:0.12.18`.

> localstack0.12.18 | 2021-09-29T10:08:37:INFO:localstack.services.generic_proxy: Blocked cors request from forbidden origin http://127.0.0.1:8000

I have the following cors set on the bucket:

```json
{
    "CORSRules": [
        {
            "AllowedOrigins": ["http://127.0.0.1:8080", "http://localhost:8080", "http://127.0.0.1:8000", "http://localhost:8000"],
            "AllowedMethods": ["GET", "HEAD", "PUT", "POST", "DELETE"],
            "MaxAgeSeconds": 3000,
            "ExposeHeaders": ["ETag"],
            "AllowedHeaders": ["*"]
        }
    ]
}

```


If I use `localstack/localstack:0.12.10`, the `forbidden origin` issue doesn't occur.

**Comment by whummer** [comment]
Thanks for reporting @richardARPANET @monkut . This should be fixed in #4994 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists.. Thanks!
